static/app/components/timeRangeSelector/utils.tsx

```diff
@@ -29,7 +29,7 @@ export type RelativeUnitsMapping = {
 
 const DATE_TIME_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
 
-const STATS_PERIOD_REGEX = /^(\d+)([smhdw]{1})$/;
```

Comment:
I suspect the `RelativeUnitsMapping` should be a proper mapping as well, `Record<RelativePeriodUnit, <...shape>>`

Reply:
Now that I think about it, I wonder if the regex should be compiled dynamically from the object, otherwise they're destined to be out of sync forever

Reply:
probably, but I don't know how many more time periods we'll ever be adding, maybe second or month tops.

Reply:
Am now seeing that there are multiple `RelativeUnitsMapping` types, multiple `SUPPORTED_RELATIVE_PERIOD_UNITS` declarations, etc. I'll clean that all up separately 😞